### ISSUE TITLE
Bookshelf on Kubernetes Engine Tutorial: Backend doesn't start

### ISSUE THREAD

**Issue by dhruvbhatia** [GoogleCloudPlatform/getting-started-ruby#87]
I'm trying to [follow the Bookshelf on Kubernetes Engine tutorial](https://cloud.google.com/ruby/tutorials/bookshelf-on-kubernetes-engine), however get caught out when trying to deploy the `bookshelf-worker` backend, [as per this step](https://cloud.google.com/ruby/tutorials/bookshelf-on-kubernetes-engine#deploying_the_bookshelf_back_end).

It looks like the `bookshelf-worker` pods fail to start and then backs off after too many failures:
```sh
> kubectl create -f bookshelf-worker.yaml
> kubectl get pods

NAME                                  READY     STATUS             RESTARTS   AGE
bookshelf-frontend-4221472564-czvth   1/1       Running            0          13m
bookshelf-frontend-4221472564-fwkbz   1/1       Running            0          13m
bookshelf-frontend-4221472564-vjmzg   1/1       Running            0          13m
bookshelf-worker-3385195378-xg9r5     0/1       CrashLoopBackOff   6          8m
bookshelf-worker-3385195378-zqwx9     0/1       CrashLoopBackOff   6          8m
```

I looked at the logs and a `NoMethodError` seems to be the culprit:

```bash
> kubectl logs -f bookshelf-worker-3385195378-xg9r5
12:26:22 worker.1 | started with pid 31
12:26:25 worker.1 | Running worker to lookup book details
12:26:26 worker.1 | rake aborted!
12:26:26 worker.1 | NoMethodError: undefined method `subscription' for nil:NilClass
12:26:26 worker.1 | /app/lib/active_job/queue_adapters/pub_sub_queue_adapter.rb:32:in `run_worker!'
12:26:26 worker.1 | /app/lib/tasks/run_worker.rake:4:in `block in <top (required)>'
12:26:26 worker.1 | /app/vendor/bundle/ruby/2.2.0/gems/rake-12.3.0/exe/rake:27:in `<top (required)>'
12:26:26 worker.1 | Tasks: TOP => run_worker
12:26:26 worker.1 | (See full trace by running task with --trace)
12:26:26 worker.1 | exited with code 1
12:26:26 system   | sending SIGTERM to all processes
```

Any ideas on how to resolve? Assistance appreciated.

**Comment by frankyn** [comment]
Hi @dhruvbhatia,

Thanks for filing an issue. Follow-up question, did you create a Pub/Sub topic named `lookup_book_details_queue`? This might be the issue given the error message.

**Comment by dhruvbhatia** [comment]
Thanks @frankyn.
I tried creating the `lookup_book_details_queue` topic as per your advice, however I'm unfortunately still seeing the same error message. Any other ideas?

**Comment by frankyn** [comment]
I missed the step of also creating a PULL subscription for `lookup_book_details_queue` topic named `lookup_book_details`. I'm hitting another bug in the frontend. I'll update this issue when I have a proper fix.

**Comment by frankyn** [comment]
@dhruvbhatia I created a short patch to help fix the follow-up issue in [fix.patch](https://gist.github.com/frankyn/a24b4b9567e1eabb6fab993d0c5f7b0c).

You can apply it to the cloned repo by using:

```
git apply fix.patch
```

It should allow you to start it successfully given the topic `lookup_book_details_queue`.

Please let me know if this unblocks you, thank you!

**Comment by frankyn** [comment]
Friendly bump, @dhruvbhatia. Did this resolve your issue?

**Comment by dhruvbhatia** [comment]
Hi @frankyn, thanks again. I saved that gist as `fix.patch`, however I'm getting the following error when trying trying to `git apply fix.patch`:
```sh
error: corrupt patch at line 200
```

**Comment by frankyn** [comment]
@dhruvbhatia, ah I didn't modify something correctly. I updated the gist could you try again?

**Comment by dhruvbhatia** [comment]
@frankyn thanks. Unfortunately still receiving the `corrupt patch` message when trying to apply it! Any other ideas?

**Comment by frankyn** [comment]
I'm working on an update to the repo @dhruvbhatia, if you'd like the fix before I merge use:

```
git clone git@github.com:GoogleCloudPlatform/getting-started-ruby.git
git checkout fix-kubernetes-engine
```
We appreciate your patience. Apologies for the delays.
